### PR TITLE
Updating FileSystem.AccessControl examples

### PR DESCRIPTION
**Pull request by carlossanlop**
- Make sure the samples are using the latest .NET 5.0 features
- Move them to sample files instead of having them directly written in the remarks

### CODE REVIEW

**Review by gewarren — APPROVED** [review]
LGTM - [preview link](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.filesystemaclextensions.create?view=net-5.0&branch=pr-en-us-5247#System_IO_FileSystemAclExtensions_Create_System_IO_DirectoryInfo_System_Security_AccessControl_DirectorySecurity_).

File: samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/DirectoryInfoCreateDirectory.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public class MyClassCS
+    {
+        // Attribute to address CA1416 warning:
+        // System.IO.FileSystem.AccessControl APIs are  only available on Windows
+        [SupportedOSPlatform("windows")]
+        static void Main()
+        {
+            // Create the file security object
+
+            SecurityIdentifier identity = new SecurityIdentifier(WellKnownSidType.BuiltinUsersSid, null);
```
Comment (adamsitnik) [inline samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/DirectoryInfoCreateDirectory.cs:17]: I would add a comment explaining what `BuiltinUsersSid` is and why the 2nd parameter can be null. 

```suggestion
            SecurityIdentifier identity = new SecurityIdentifier(
                WellKnownSidType.BuiltinUsersSid, // this maps to "Everyone" user group in Windows
                null); // null is OK for this particualr user group, for others a non empty value might be required
```

File: samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    Console.WriteLine(encoding.GetString(readBuffer));
+
+                    /*
+                        Output:
+                            Hello world!
+                    */
+                }
+            }
+        }
```
Comment (adamsitnik) [inline samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs:66]: the description of this example says:

> The following code example creates a new text file inside the user's temporary folder, explicitly specifying all security attributes

we should rather focus on the security aspect here (passing `FileSecurity` to the right method) and just create the file, without writing and reading the "Hello World". It just adds a lot of code to the example and makes it more complex and harder to understand. Just keep it simple.

File: samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/DirectoryInfoCreateDirectory.cs (hunk 1 above)
Comment (carlossanlop) [reply samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/DirectoryInfoCreateDirectory.cs:17]: Sounds good. I'll add it. I adjusted a small typo.

File: samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs (hunk 2 above)
Comment (carlossanlop) [reply samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs:66]: I see what you mean. I wanted to show that the file was still readable after altering the file security. But sounds more like something we would do in a unit test, not in an example.

File: samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                while (stream.Read(readBuffer, 0, readBuffer.Length) > 0)
+                {
+                    Console.WriteLine(encoding.GetString(readBuffer));
+
+                    /*
+                        Output:
+                            Hello world!
+                    */
+                }
+            }
```
Comment (carlossanlop) [inline samples/snippets/csharp/VS_Snippets_CLR/IO.FileSystem.AccessControl/FileInfoCreate.cs:65]:
```suggestion
            } // Dispose flushes
```

**Pull request merged**